### PR TITLE
feat(core): add a fallback mechanism for avatar

### LINKED ISSUE #5984: Add a fallback mechanism for avatar
**What is the problem that this feature will solve?**
If I want to display an avatar, I want to have a fallback mechanism in case something fails.

**Who will benefit from this feature?**
Every team which uses the fundamental library. 

**Describe the solution you'd like**
If there is an error loading the avatar image, the component falls back to an alternative in a specific order:
1. the provided children
2. the first letter of the alt text or label
3. a generic avatar icon

Examples:
< fd-avatar alt="one"  image="https://brokenImage..." label="Jane Doe">B< /fd-avatar>
< fd-avatar alt="R"  image="https://brokenImage..." label="Jane Doe">< /fd-avatar>
< fd-avatar image="https://brokenImage...">< /fd-avatar>

Leads to:
Screen Shot 2021-07-14 at 1 37 07 PM

### PR DESCRIPTION
**Pull request by aberikashvili**
#### Please provide a link to the associated issue.
Closes #5984

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/avatar/avatar-docs.component.html [hunk 1]
```diff
@@ -132,3 +132,33 @@
...
+
+<fd-docs-section-title id="default-image-alter-options" componentName="avatar">
+    Default image with alter icon options
+</fd-docs-section-title>
+<description>
+    To provide alternative icon options by priority use the <code>[alterIcon]</code> property. <br>
+    There's few options can be set the order priority, Avatar component will process them and display first successful option if <code>[image]</code> url fails to load: 
+    <ul>
+        <li>
+            <code>contet</code> - Avatar will display first letter from string, provided as component content
```
Comment (mikerodonnell89) [inline apps/docs/src/app/core/component-docs/avatar/avatar-docs.component.html:146]: `<code>contet</code>` typo should be `<code>content</code>`

File: apps/docs/src/app/core/component-docs/avatar/avatar-docs.component.html [hunk 2]
```diff
@@ -132,3 +132,33 @@
...
+        </li>
+        <li>
+            <code>alt</code> - Avatar will display first letter from string, provided as <code>alt</code> attribute
+        </li>
+        <li>
+            <code>custom</code> - Avatar will display image, provided as <code>defaultImage</code>
```
Comment (mikerodonnell89) [inline apps/docs/src/app/core/component-docs/avatar/avatar-docs.component.html:152]: This part is confusing, let's have it be <code>backupImage</code> - Avatar will display image, provided as <code>backupImage</code>

File: libs/core/src/lib/avatar/avatar.component.ts [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
     ViewEncapsulation
 } from '@angular/core';
 import { ANY_LANGUAGE_LETTERS_REGEX, ColorAccent, Size, applyCssClass, getRandomColorAccent, CssClassBuilder } from '@fundamental-ngx/core/utils';
 
 let avatarUniqueId = 0;
 
+const ALTER_ICON_OPTIONS = {
+    CONTENT: 'content',
+    ALT: 'alt',
+    CUSTOM: 'custom',
```
Comment (mikerodonnell89) [inline libs/core/src/lib/avatar/avatar.component.ts:20]: change to backupImage?

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Looks good! Couple change requests

File: libs/core/src/lib/avatar/avatar.component.ts (hunk 3 above)
Comment (aberikashvili) [reply libs/core/src/lib/avatar/avatar.component.ts:20]: Done

File: apps/docs/src/app/core/component-docs/avatar/avatar-docs.component.html (hunk 2 above)
Comment (aberikashvili) [reply apps/docs/src/app/core/component-docs/avatar/avatar-docs.component.html:152]: Done

File: apps/docs/src/app/core/component-docs/avatar/avatar-docs.component.html (hunk 1 above)
Comment (aberikashvili) [reply apps/docs/src/app/core/component-docs/avatar/avatar-docs.component.html]: Done

**Review by olshansky — APPROVED** [review]

**Review by N1XUS — APPROVED** [review]

**Review by DeepakSap14 — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**